src/zkscript/merkle_trees/merkle_tree.py

```diff
@@ -0,0 +1,198 @@
+from tx_engine import Script
+
+
+class MerkleTree:
+    """Class representing a Merkle Tree.
+
+    This class takes a root hash and a specified hash function to create the Merkle Tree
+    structure.
+
+    Attributes:
+        root (str): The root hash of the Merkle Tree. Must be a valid hexadecimal value.
+        hash_function (str): The hash function used in the Merkle Tree. This must be a valid opcode hash function or a
+            combination of valid opcode hash functions.
+        depth (int): The depth (number of levels) of the Merkle tree.
+
+    Methods:
```

Comment:
Can we agree that the doc of classes does not need to contain the doc of methods? Instead, this doc could be moved to the __init__ directly. Same thing for attributes. For attributes, you could add the pre-conditions in a Pre-conditions section of the documentation.

Reply:
fixed!

Reply:
fixed!